src/c/backend/task.cpp

```diff
@@ -67,18 +71,25 @@ bool InnerTask::add_dependency(InnerTask *task) {
 
   bool dependency_complete = false;
 
-  if (task->add_dependent(this)) {
-    this->num_blocking_dependencies--;
-
-    // Return that the dependency was already complete
-    dependency_complete = true;
+  Task::State dependent_state = task->add_dependent(this);
+
+  if (dependent_state >= Task::dispatched) {
+    this->num_blocking_dependencies.fetch_sub(1);
+    this->num_unspawned_dependencies.fetch_sub(1);
```

Comment:
Is this guaranteed that this counter is always increased by 1 for each dependency? (even though dependency was correctly spawned before dependent tasks are spawned)

Reply:
I think so (L108~111)

Reply:
At the moment, yes. It'll need some changes once we have other phases calling "notify"-like parts.